cpp/src/groupby/hash/groupby.cu

```diff
@@ -646,18 +646,18 @@ std::unique_ptr<table> groupby(table_view const& keys,
 bool can_use_hash_groupby(host_span<aggregation_request const> requests)
 {
   return std::all_of(requests.begin(), requests.end(), [](aggregation_request const& r) {
-    // Currently, structs are not supported in any of hash-based aggregations.
-    // Therefore, if any request contains structs then we must fallback to sort-based aggregations.
-    // TODO: Support structs in hash-based aggregations.
     auto const v_type = is_dictionary(r.values.type())
                           ? cudf::dictionary_column_view(r.values).keys().type()
                           : r.values.type();
 
-    return not(r.values.type().id() == type_id::STRUCT) and
-           std::all_of(r.aggregations.begin(), r.aggregations.end(), [v_type](auto const& a) {
-             return cudf::has_atomic_support(cudf::detail::target_type(v_type, a->kind)) and
-                    is_hash_aggregation(a->kind);
-           });
+    // Currently, structs and lists values are not supported in any of hash-based aggregations.
+    // For those situations, we must fallback to sort-based aggregations.
```

Comment:
This is not true anymore, is it?

Reply:
This is true. Technically we can implement code to support hash-based aggregates for list type. However, we didn't implement any yet.

Reply:
I'm confused, how do we run algorithms like `distinct` which hash key based columns? `distinct` supports list type as key

Reply:
Yeah but here we are checking for lists as **values**. We support lists as keys, which is a different story.

Reply:
I was confused by that as well, on the first read. I skipped over the "*values*" part.
Would there be value in including your clarification in the comment?
> lists as values. We support lists as keys, which is a different story.

Reply:
Thanks. I've updated the comments.